### PR TITLE
Integrate PostHog into n8n

### PR DESCRIPTION
**Pull request by ivov**

### CODE REVIEW

File: packages/cli/src/ExternalHooks.ts [hunk 1]
```diff
@@ -38,6 +38,9 @@ class ExternalHooksClass implements IExternalHooksClass {
 
 		// Load all the provided hook-files
 		for (let hookFilePath of externalHookFiles) {
+			// eslint-disable-next-line no-continue
+			if (hookFilePath.includes('posthog')) continue; // FE only
+
```
Comment (ahsanv) [inline packages/cli/src/ExternalHooks.ts:43]:
```suggestion
```

Comment (ahsanv) [reply packages/cli/src/ExternalHooks.ts:43]: not needed anymore

File: packages/cli/src/telemetry/index.ts [hunk 2]
```diff
@@ -122,8 +126,10 @@ export class Telemetry {
 		clearInterval(this.pulseIntervalReference);
 		void this.track('User instance stopped');
 		return new Promise<void>((resolve) => {
-			if (this.client) {
-				this.client.flush(resolve);
+			if (this.postHog) this.postHog.shutdown();
```
Comment (ahsanv) [inline packages/cli/src/telemetry/index.ts:129]: nitpick: over time i've come to prefer consistency in blocks, wdyt?
```suggestion
			if (this.postHog) {
				this.postHog.shutdown();
			}
```

File: packages/editor-ui/src/components/Telemetry.vue [hunk 3]
```diff
@@ -26,28 +29,36 @@ export default Vue.extend({
...
-				const instanceId = this.$store.getters.instanceId;
-				const userId = this.$store.getters['users/currentUserId'];
-				const logLevel = this.$store.getters['settings/logLevel'];
-				this.$telemetry.init(opts, { instanceId, logLevel, userId, store: this.$store });
-			}
+			if (this.isTelemetryInitialized || !this.isTelemetryEnabledOnRoute) return;
+
+			const telemetrySettings = this.telemetry;
+
+			if (!telemetrySettings || !telemetrySettings.enabled) return;
```
Comment (ahsanv) [inline packages/editor-ui/src/components/Telemetry.vue:36]: nitpick:
```suggestion
			if (!telemetrySettings || !telemetrySettings.enabled) {
				return;
			}
```

**Review by ahsanv — COMMENTED** [review]
Looks good to me! Minor comments and nitpicks. Approval pending final testing.

File: packages/cli/src/telemetry/index.ts (hunk 2 above)
Comment (ivov) [reply packages/cli/src/telemetry/index.ts]: I find one-liners more readable but not precious about it. Changed.

File: packages/editor-ui/src/components/Telemetry.vue (hunk 3 above)
Comment (ivov) [reply packages/editor-ui/src/components/Telemetry.vue]: Changed.

**Review by ahsanv — APPROVED** [review]
🎉

**Comment by janober** [comment]
Got released with `n8n@0.192.0`

**Pull request merged**